HSNE/src/HsneAnalysisPlugin.cpp

```diff
@@ -203,70 +219,74 @@ void HsneAnalysisPlugin::computeTopLevelEmbedding()
     // Number of landmarks on the top scale
     int numLandmarks = topScale.size();
 
-    // Create a subset of the points corresponding to the top level HSNE landmarks,
-    // Then create an empty embedding derived from this subset
-    auto inputDataset       = getInputDataset<Points>();
-    auto selectionDataset   = inputDataset->getSelection<Points>();
-
-    // Select the appropriate points to create a subset from
-    selectionDataset->indices.resize(numLandmarks);
-
-    if (inputDataset->isFull())
-    {
-        for (int i = 0; i < numLandmarks; i++)
-            selectionDataset->indices[i] = topScale._landmark_to_original_data_idx[i];
-    }
-    else
+    // Only create new selection helper if it does not exist yet
+    if (!_selectionHelperData.isValid())
```

Comment:
I didn't change any of the following lines expect making use of the new member `_selectionHelperData` instead of the local `mv::Dataset<Points> subset` variable.
The diff tool is just confused that all the code moved into an if-block